### DISCUSSION TITLE
allow css to identify a specific dialog, e.g. emoji dialog

### DISCUSSION THREAD

**Discussion by scil** [tinymce/tinymce#8230] [Show and tell]
many different plugins use the tinymce model dialog, but in theme silver, there is no unique id for there different dialog.

I only found there only different `title` in the process of renderring `.tox-dialog__header`, so I tried to keep the `title` in the `.tox-dialog__header`,  then exact the `title` into the class list of the dialog container.

there is the path file
```
diff --git a/node_modules/tinymce/themes/silver/theme.js b/node_modules/tinymce/themes/silver/theme.js
index 72cf12f..ef8fd88 100644
--- a/node_modules/tinymce/themes/silver/theme.js
+++ b/node_modules/tinymce/themes/silver/theme.js
@@ -31539,7 +31539,7 @@
       var pClose = ModalDialog.parts.close(renderClose(providersBackstage));
       var components = [pTitle].concat(spec.draggable ? [pHandle] : []).concat([pClose]);
       return Container.sketch({
-        dom: fromHtml('<div class="tox-dialog__header"></div>'),
+        dom: fromHtml('<div class="tox-dialog__header ' + spec.title + '"></div>'), //scil
         components: components
       });
     };
@@ -31584,6 +31584,11 @@
       };
     };
     var renderModalDialog = function (spec, initialData, dialogEvents, backstage) {
+      //scil
+      //debugger
+      //console.debug('renderModalDialog', spec);
+      spec.extraClasses = [ spec.header.dom.classes[1] ];
+
       var _a;
       var updateState = function (_comp, incoming) {
         return Optional.some(incoming);

```

then, you can custom the specific dialog of specific plugin, e.g. emoji dialog:
```
.tox-dialog.Emoticons .tox-dialog__body-nav,
.tox-dialog.表情           .tox-dialog__body-nav,
.tox-dialog.Emoticons .tox-form__group:first-child,
.tox-dialog.表情           .tox-form__group:first-child {
  display: none;
}

```

the `title` varies on different languages.

**Comment by lostkeys** [comment]
This was a deliberate design decision to avoid ugly hacks that caused a lot of support overhead for us and also upgrade difficulties. Sometimes the HTML and CSS changes between minor and patch releases and it can potentially break customer's editors running on our cloud version that auto-updates. To help us understand your use-case, what are you trying to achieve?
— Fredrik, Product Manager

**Reply by scil** [reply]
Thank you.

I just want to simplify the emoji dialog by using css 'display:none' , meanwhile keeping other dialogs not affected.

**Reply by lostkeys** [reply]
Sorry for not getting back to you earlier, what part of the dialog do you want to simplify and why?

Thanks,

**Reply by scil** [reply]
emoji dialog.

it holds so many elements, including search box and some tabs.  I would like a simple emoji dialog for my users, just like here, github.

**Comment by ricardo-reis-1970** [comment]
Dear moderator,

I just got an email from `MichelleCarter3596 <notifications@github.com>` regarding this issue #8230, with links to (DO NOT CLICK! NSFW!):
- `<censored by moderator>`
- `<censored by moderator>`

These are _pornados_ with links to, among others, `<censored by moderator>`.

These could come from a legitimate user whose computer / github account was hacked, or from a user that should not be legitimate. The only **Michelle Carter** I could find on GitHub is a frontend engineer with Adobe and I'm dead sure she's not one of the ladies in the website.

Is there a way this could be tracked on your end?

**Reply by lostkeys** [reply]
I got the email notification too this morning (for several threads) but they are gone on GitHub so they probably have some spam filter kicking in. A bit annoying to get the emails but little we can do.

Cheers,
Fredrik, Tiny

**Reply by TheSpyder** [reply]
In future I suggest not repeating the links in your report, as then they become permanent rather than removed by github's filtering mechanisms. Or, in the worst case, you may become flagged as a spammer yourself.
